.github/workflows/java-model-diff.yml

```diff
@@ -0,0 +1,98 @@
+name: Models as Data
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+    inputs:
+      projects:
+        description: "The projects to generate models for"
+        required: true
+        default: '["netty/netty"]'
+  pull_request:
+    branches:
+      - main
+    paths:
+      - "java/ql/src/utils/model-generator/**/*.*"
+
+jobs:
+  model-diff:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        # large
+        # ["google/guava", "spring-projects/spring-framework", "apache/poi"]
+        # others
+        # ["FasterXML/jackson-core", "FasterXML/jackson-databind", "google/gson", "JodaOrg/joda-time"
+        slugs: ${{fromJson(github.event.inputs.projects || '["apache/commons-codec", "apache/commons-io", "apache/commons-beanutils", "apache/commons-logging", "apache/commons-fileupload", "apache/commons-lang", "apache/commons-validator", "apache/commons-csv", "apache/dubbo"]' )}}
+    steps:
+      - name: Clone self (github/codeql) for baseline
+        uses: actions/checkout@v2
+        with:
+          path: codeql-baseline
+          ref: ${{ github.base_ref }}
+      - name: Clone self (github/codeql) with new generator
+        uses: actions/checkout@v2
+        with:
+          path: codeql-head
+          ref: ${{ github.ref }}
+      - uses: ./codeql-baseline/.github/actions/fetch-codeql
+      - name: Download database
+        env:
+          SLUG: ${{ matrix.slugs }}
+        run: |
+          set -x
+          mkdir lib-dbs
+          SHORTNAME=${SLUG//[^a-zA-Z0-9_]/}
+          projectId=`curl -s https://lgtm.com/api/v1.0/projects/g/$SLUG | jq .id`
+          curl -L "https://lgtm.com/api/v1.0/snapshots/$projectId/java" -o $SHORTNAME.zip
+          unzip -q -d $SHORTNAME-db $SHORTNAME.zip
+          mkdir lib-dbs/$SHORTNAME/
+          mv $SHORTNAME-db/`ls -1 $SHORTNAME-db`/* lib-dbs/$SHORTNAME/
```

Comment:
General recommendation: double-quote everywhere you're using variables or names controlled by the external file. Your replacement will ensure the short name won't have spaces, but it's good to be defensive.

I prefer to use `"$(...)"` over `` `...` `` to handle spaces in the output, but that is partly personal style.

Reply:
Also why do you have to unzip and move rather than unzipping straight into `lib-dbs/$SHORTNAME`?

Reply:
This is to handle the fact that we don't know the root folder of the zip archive (due to the naming pattern). With this indirection, we can move it to a known place (`lib-dbs/$SHORTNAME`)